packages/plugin-kafka/src/kafka.ts

```diff
@@ -0,0 +1,87 @@
+import assert = require('assert')
+import { resolve } from 'path'
+import { NotFoundError } from 'common-errors'
+import { Microfleet, PluginTypes, LoggerPlugin, PluginInterface } from '@microfleet/core'
+// import retry = require('bluebird-retry')
+import * as Kafka from 'node-rdkafka'
+
+/**
+ * Relative priority inside the same plugin group type
+ */
+export const priority = 0
+export const name = 'kafka'
+export const type = PluginTypes.transport
+
+export type TopicConfig = {
+  topicName: string;
+}
+
+type GlobalConfig = {
+  'metadata.broker.list': string;
+}
+
+type Config = {
+  globalConf: GlobalConfig;
+  topicConf?: TopicConfig;
+}
+
+/**
+ * Defines service extension
+ */
+export interface KafkaPlugin {
+  globalConf: GlobalConfig
+  createConsumer: (topicConf?: TopicConfig) => Kafka.KafkaConsumer
+  createProducer: (topicConf?: TopicConfig) => Kafka.Producer
+}
+
+export class KafkaFactory implements KafkaPlugin {
```

Comment:
great work, next step would be to actually implement handling of events so that we get a read / write stream of sorts with iterators for a nice and useful API

you can watch more about it here - https://www.youtube.com/watch?v=oc4PpG01Upw&list=PL0CdgOSSGlBbkVfya8-yyhVsaiTx2GKXF&index=13&t=0s